### PR TITLE
ISPN-7680 - enabled karaf-maven-plugin to validate Karaf features

### PR DESCRIPTION
**Pull request by paoloantinori**

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -544,6 +560,40 @@
                   <compilerArgument>-AtranslationFilesPath=${project.basedir}/target/generated-translation-files</compilerArgument>
                </configuration>
             </plugin>
+            <plugin>
+               <groupId>org.apache.karaf.tooling</groupId>
+               <artifactId>karaf-maven-plugin</artifactId>
+               <version>${version.karaf-maven-plugin}</version>
```
Comment (danberindei) [inline bom/pom.xml]: I think this should be in the parent POM, that's where we usually put the plugins.

File: core/pom.xml [hunk 2]
```diff
@@ -209,6 +215,15 @@
                </execution>
             </executions>
          </plugin>
+         <plugin>
+            <groupId>org.apache.karaf.tooling</groupId>
+            <artifactId>karaf-maven-plugin</artifactId>
```
Comment (danberindei) [inline core/pom.xml]: Please move this before the defaults plugin, since the bundle plugin, the helper and the karaf plugin all seem related.

**Comment by slaskawi** [comment]
@paoloantinori any update?

File: all/embedded/pom.xml [hunk 3]
```diff
@@ -134,6 +134,13 @@
       </dependency>
 
       <dependency>
+        <groupId>org.apache.karaf.features</groupId>
+        <artifactId>framework</artifactId>
+        <type>kar</type>
+        <scope>provided</scope>
```
Comment (paoloantinori) [inline all/embedded/pom.xml]: Yes, this is probably not needed

**Comment by galderz** [comment]
Hmm, tried to resolve conflicts myself but not sure this has worked. Let me close the PR and send a new one with the conflicts fixed.

**Comment by galderz** [comment]
Submitted new PR fixing conflicts #5117

**Pull request closed without merging**